mage_ai/data_preparation/models/pipeline.py

```diff
@@ -198,6 +207,54 @@ def create(self, name, pipeline_type=PipelineType.PYTHON, repo_path=None):
         )
         return pipeline
 
+    @classmethod
+    def import_from_zip(self, zip_content, overwrite=False):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            zip_data = BytesIO(zip_content)
+            with zipfile.ZipFile(zip_data, 'r') as zipf:
+                zip_size = sum(e.file_size for e in zipf.infolist())  # calc zip size in bytes
+                if zip_size / 1000 > PIPELINE_MAX_SIZE:  # prevention against zip-bombs
+                    raise PipelineZipTooLargeError(
+                        f'Pipeline zip excedes size limit {PIPELINE_MAX_SIZE/1000}Kb')
+
+                zip_contents = zipf.namelist()
+                prefix = os.path.commonpath(zip_contents)  # Check if zip contents are part of a root folder
+                
+                zipf.extractall(tmp_dir)
+                if prefix:
+                    for file in zip_contents[1:]:  # move each file except the folder itself
+                        original_path = os.path.join(tmp_dir, file)
+                        final_path = os.path.join(tmp_dir, os.path.relpath(file, prefix))
+                        os.rename(original_path, final_path)
+                    os.removedirs(os.path.join(tmp_dir, prefix))  # remove root folder
+
+            config_zip_path = self.__find_pipeline_file(
+                self, tmp_dir, PIPELINE_CONFIG_FILE, PIPELINES_FOLDER
+            )
+            if config_zip_path is None or not os.path.exists(config_zip_path):
+                raise InvalidPipelineZipError
+
+            p_files, conf_dest_path, p_name = self.__update_pipeline_yaml(
+                self,
+                tmp_dir,
+                config_zip_path,
+                overwrite,
+            )
+
+            # write all files in bulk
+            for source, destination in p_files:
+                try:
+                    dir_path, file_name = os.path.split(destination)
+                    with open(source, 'r') as src:
+                        File.create(file_name, dir_path, src.read(), overwrite=overwrite)
+                except Exception:
+                    raise FileWriteError(f'Failed to write pipeline file to {destination}.')
+
+            # return the pipelin configuration file
+            ret_file = File.from_path(conf_dest_path)
+            ret_file.filename = p_name
```

Comment:
why are we setting the `filename` here?

Reply:
Since I'm using the file creation framework to also import pipelines, a returned file is expected. I decided to return the pipeline configuration file but not before modifying its name from 'metadata.yaml' to the pipeline's name. The app then takes this name and displays it on the UI after the import process takes place.